src/axom/primal/geometry/BezierCurve.hpp

```diff
@@ -276,22 +287,76 @@ class BezierCurve
     m_weights[idx] = weight;
   };
 
-  /// Checks equality of two Bezier Curve
-  friend inline bool operator==(const BezierCurve<T, NDIMS>& lhs, const BezierCurve<T, NDIMS>& rhs)
+  /// Returns a copy of the Bezier curve's weights
+  WeightsVec getWeights() const { return m_weights; }
+
+  /// Returns an axis-aligned bounding box containing the Bezier curve
+  BoundingBoxType boundingBox() const
   {
-    return (lhs.m_controlPoints == rhs.m_controlPoints) && (lhs.m_weights == rhs.m_weights);
+    return BoundingBoxType(m_controlPoints.data(), static_cast<int>(m_controlPoints.size()));
   }
 
-  friend inline bool operator!=(const BezierCurve<T, NDIMS>& lhs, const BezierCurve<T, NDIMS>& rhs)
+  /// Returns an oriented bounding box containing the Bezier curve
+  OrientedBoundingBoxType orientedBoundingBox() const
   {
-    return !(lhs == rhs);
+    return OrientedBoundingBoxType(m_controlPoints.data(), static_cast<int>(m_controlPoints.size()));
   }
 
-  /// Returns a copy of the Bezier curve's control points
-  CoordsVec getControlPoints() const { return m_controlPoints; }
+  /*!
+   * \brief Predicate to check if the Bezier curve is approximately linear
+   *
+   * This function checks if the internal control points of the BezierCurve
+   * are approximately on the line defined by its two endpoints
+   *
+   * \param [in] tol Threshold for sum of squared distances
+   * \param [in] useStrictLinear If true, checks that the control points are
+   *   evenly spaced along the line
```

Comment:
```suggestion
   *   evenly spaced along the line AND not too far from the line
```

That detail is in the internal comment and seems important enough to include here.